src/core_tests.rs

```diff
@@ -431,37 +408,13 @@ fn sort_nodes_by_distance_to(nodes: &mut [TestNode], name: &XorName) {
 }
 
 /// Verify that the invariant is upheld for the node at `index`.
-fn verify_invariant_for_node(nodes: &[TestNode], index: usize) {
-    let routing_tables = nodes.iter().map(|node| node.routing_table()).collect_vec();
-    verify_invariant(&routing_tables, index);
+fn verify_invariant_for_node(node: &TestNode) {
+    node.routing_table().verify_invariant();
 }
 
-/// Verify that the invariant is upheld for the routing table at `index`.
-pub fn verify_invariant(_routing_tables: &[RoutingTable<XorName>], _index: usize) {
-    // let target = routing_tables[index].our_name();
-    // let mut count = routing_tables.len() - 1;
-    // let mut bucket_index = 0;
-
-    // while count > 0 {
-    //     let actual_bucket = actual_names_in_bucket(&routing_tables[index], bucket_index);
-    //     let expected_bucket = expected_names_in_bucket(routing_tables, target, bucket_index);
-    //     if actual_bucket.len() < MIN_GROUP_SIZE {
-    //         assert!(expected_bucket == actual_bucket,
-    //                 "Node: {:?}, expected: {:?}. found: {:?}",
-    //                 target,
-    //                 expected_bucket,
-    //                 actual_bucket);
-    //     }
-    //     count -= expected_bucket.len();
-    //     bucket_index += 1;
-    // }
-}
-
-/// Verify that the invariant is upheld for all nodes.
 fn verify_invariant_for_all_nodes(nodes: &[TestNode]) {
-    for node_index in 0..nodes.len() {
-        verify_invariant_for_node(nodes, node_index);
-    }
+    let routing_tables: Vec<RoutingTable<_>> = nodes.iter().map(TestNode::routing_table).collect();
```

Comment:
Maybe slightly less verbose would be to use `collect_vec()` from itertools rather than `collect()`?